# SKEETER

## Relative abundance Analysis Tool

This tool processes PAF files generated by Minimap2 to analyze the abundance of different mosquito species in metagenomic samples.
This repository contains a Python script for analyzing the abundance of species using COI (Cytochrome c oxidase subunit I) sequences obtained from sequencing data. 
The script utilizes the Minimap2 tool for alignment and performs subsequent data processing and visualization using pandas and Plotly.

## Features

- **Minimap2 Integration:** Automatically runs Minimap2 on a set of input FASTQ or FASTA files and generates PAF files.
- **Abundance Calculation:** Processes PAF files to calculate the abundance of different species based on alignment data.
- **Visualization:** Generates interactive bar plots using Plotly to visualize the abundance percentages of different species across samples.

## Prerequisites

Before using this tool, ensure you have the following dependencies installed:

- Python 3
- requirements
  ```
  pip install -r requirements.txt
  ```
- Minimap2
  ```
  conda install minimap2
  ```

## Usage

1. Clone the repository:

```bash
git clone https://github.com/stockmany/SKEETER
```

2. Navigate to the Directory:
```
cd SKEETER
```
3. Install dependencies
```
pip install -r requirements.txt
```
