# SKEETER

## Relative abundance Analysis Tool

This tool processes PAF files generated by Minimap2 to analyze the abundance of different mosquito species in metagenomic samples.
This repository contains a Python script for analyzing the abundance of mosquito species using COI (Cytochrome c oxidase subunit I) sequences obtained from sequencing data. 
The script utilizes the Minimap2 tool for alignment and performs subsequent data processing and visualization using pandas and Plotly.

Abundance percentage calculations are performed using harmonic mean calculations based on the paper: [Comparative analysis of metagenomic classifers for longâ€‘read sequencing datasets](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-024-05634-8)

## Features

- **Minimap2 Integration:** Automatically runs Minimap2 on a set of input FASTQ or FASTA files and generates PAF files.
- **Abundance Calculation:** Processes PAF files to calculate the abundance of different species based on alignment data.
- **Visualization:** Generates interactive bar plots using Plotly to visualize the abundance percentages of different species across samples.

## Installation
1. Create conda environment.
```bash
conda create -n env_name python=3.12.* pandas plotly minimap2 -c bioconda
```
2. Clone the repository:
```bash
git clone https://github.com/stockmany/SKEETER
```

## Usage
```bash
python SKEETER/P2RA.py -i [input_folder/file] -o [output_directory]
```

optional arguments:
  -h, --help            show this help message and exit
  
optional: add alias for easier command
```bash
echo 'alias Skeeter="~/SKEETER/P2RA.py" '>> ~/.bashrc
```

## Steps involved
**Alignment with Minimap2:**
Minimap2 is used to align sequencing reads against a reference COI sequence database (COI_database.fasta).

**Data Processing:**
The script processes the PAF (Pairwise mApping Format) files generated by Minimap2.
It calculates the harmonic mean of alignment length and number of matches for each query.

**Abundance Calculation:**
The script calculates the relative abundance of each species based on the alignment data.
Species with counts less than 1 are grouped as "other <1%" category.

**Data Visualization:**
The abundance data is visualized using Plotly, generating a stacked bar plot showing abundance percentages for each species across different samples.

## Input requirements
**Sequencing Data:** FASTQ and/or FASTA files containing sequencing reads.

**Reference Database:** The script utilizes the internal COI_database.fasta file for alignment.

## Output
**Abundance Data CSV**: Contains abundance percentages for each species across different samples.

**Abundance Plot HTML:** Stacked bar plot visualizing abundance percentages for each species.

## Additional notes
- Ensure that Minimap2 is installed and accessible in your system's PATH.
- The script automatically creates output directories (output, output_1, output_2, ...) to store generated files.
